searx/engines/dailymotion.py

```diff
@@ -54,45 +106,61 @@ def response(resp):
 
     # parse results
     for res in search_res['list']:
+
         title = res['title']
         url = res['url']
-        content = html_to_text(res['description'])
-        thumbnail = res['thumbnail_360_url']
+        video_id = res['id']
+
+        content = res['description']
+        if content:
+            content = content.replace('\\r', '')
+            document = lxml.html.document_fromstring(content)
+            content = " ".join(lxml.etree.XPath("//text()")(document))
+
+        if len(content) > 300:
+            content = content[:300] + '...'
+
         publishedDate = datetime.fromtimestamp(res['created_time'], None)
 
+        length = time.gmtime(res.get('duration'))
+        if length.tm_hour:
+            length = time.strftime("%H:%M:%S", length)
+        else:
+            length = time.strftime("%M:%S", length)
+
+        thumbnail = res['thumbnail_360_url']
         # http to https
         thumbnail = thumbnail.replace("http://", "https://")
 
-        results.append(
-            {
-                'template': 'videos.html',
-                'url': url,
-                'title': title,
-                'content': content,
-                'publishedDate': publishedDate,
-                'iframe_src': "https://www.dailymotion.com/embed/video/" + res['id'],
-                'thumbnail': thumbnail,
-            }
-        )
+        item = {
+            'template': 'videos.html',
+            'url': url,
+            'title': title,
+            'content': content,
+            'publishedDate': publishedDate,
+            'length': length,
+            'thumbnail': thumbnail,
+        }
+
+        # HINT: no mater what the value is, without API token videos can't shown
+        # embedded
+        allow_embed = res['allow_embed']
+        if allow_embed:
+            item['iframe_src'] = "https://www.dailymotion.com/embed/video/" + video_id
```

Comment:
Currently hard coded URLs are set at the top of the files (here `iframe_url`, see qwant.py)